### ISSUE TITLE
DetectionModelTrainer obj['xmax'] - obj['xmin']) / float(max_anchor.xmax)) # t_w ZeroDivisionError: float division by zero

### ISSUE THREAD

**Issue by lovisio** [OlafenwaMoses/ImageAI#674]
Hello,
I use this https://imageai.readthedocs.io/en/latest/customdetection/index.html to create a new detection.
Using Hololens library from example the DetectionModelTrainer working.
But using my own data generate from LabelImg, i have this error. 
"  File "C:\Python38\lib\site-packages\imageai\Detection\Custom\generator.py", line 106, in __getitem__
    w = np.log((obj['xmax'] - obj['xmin']) / float(max_anchor.xmax)) # t_w
ZeroDivisionError: float division by zero"

Do you from where to problem could be ?
My data are using only one image ( a plan ) , but with more than 200 markers on it.

**Comment by ekesdf** [comment]
Yeah I know where the problem is you have a bad annotation in your dataset where a bbox(Boundingbox) has the width 0 so you have to check all your annotation files for an object which has the same xmin and max value

**Comment by ekesdf** [comment]
Can you send in you annotation file you used for that picture rhy

**Comment by lovisio** [comment]
Thank you for your help. I saw this answer on forum , but even with only one annotation i have the problem. 
Here an example of what i use. 
[map-schema19.txt](https://github.com/OlafenwaMoses/ImageAI/files/6620354/map-schema19.txt)

**Comment by lovisio** [comment]
I tried even with just one object and i have the error.
[map-schema19.txt](https://github.com/OlafenwaMoses/ImageAI/files/6620411/map-schema19.txt)

**Comment by ekesdf** [comment]
I have found the issue :tada:

The issue is that your bboxes are to small compared to your image size. The ratio is like  0,001801106 for the x_value which is way too small so the custom anchor calculation your model anchors are [0,0,0,0] but it should be more like [50,10,150,110] for example, but tit depends on the ration between  the object and the actual image size.

So to accommodate for that you will have to cut your original images in smaller pieces, so the ratio will rise to something around 0.01 a like 832x832 pixel image would be a good size for an image tile for that size of object you want to detect

**Comment by ekesdf** [comment]
@lovisio  did it worked ?
